In [28]:
import pandas as pd

# Работа с Pandas

1. Загрузите ваш DataFrame в переменную df и выведете на экран первые пять строк 

In [29]:
df = pd.read_csv("C:/Users/Slon/PythonProjects/googleplaystore.csv")
#Переименовываем столбцы
df.rename(columns={'App':'app','Category':'category','Rating':'rating', 'Reviews':'reviews', 'Size':'size', 'Installs':'installs', 'Type':'type', 'Price':'price', 'Content Rating':'content_rating', 'Genres':'genres', 'Last Updated':'last_updated', 'Current Ver':'current_ver', 'Android Ver':'android_ver'},inplace=True)
#Выводим первые пять строк
df.head()

app        category  rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  reviews  size     installs  type  price content_rating  \
0     159   19M      10,000+  Free    0.0       Everyone   
1     967   14M     500,000+  Free    0.0       Everyone   
2   87510  8.7M   5,000,000+  Free    0.0       Everyone   
3  215644   25M  50,000,000+  Free    0.0           Teen   
4     967  2.8M     100,000+  Free    0.0       Everyone   

                      genres      last_updated         current_ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    android_ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

2. Отобразите количество строк и столбцов в DataFrame

In [30]:
df.shape

(10841, 13)

3. Выведете основную информацию о DataFrame методом info(). Проанализируйте есть ли пропуски в столбцах. Напишите вывод о том, какие столбцы содержат пропуски

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   app             10841 non-null  object 
 1   category        10841 non-null  object 
 2   rating          9367 non-null   float64
 3   reviews         10841 non-null  object 
 4   size            10841 non-null  object 
 5   installs        10841 non-null  object 
 6   type            10840 non-null  object 
 7   price           10841 non-null  float64
 8   content_rating  10840 non-null  object 
 9   genres          10841 non-null  object 
 10  last_updated    10841 non-null  object 
 11  current_ver     10833 non-null  object 
 12  android_ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


<b>Вывод:</b>

В представленном DataFrame 13 столбцов, из которых:
* 11 имеют тип `object`
* 2 имеют тип `float64`
* всего в датафрейме 10841 строка
* в столбцах 'Rating', 'Type', 'Content Rating', 'Current Ver' и 'Android Ver' есть пропуски (`NaN`)

4. Удалите или замените пропуски известными вам методами. (Подсказка: `dropna()` или `fillna()`, выбор того или иного метода должен быть аргументирован).

In [32]:
df.isna()

app  category  rating  reviews   size  installs   type  price  \
0      False     False   False    False  False     False  False  False   
1      False     False   False    False  False     False  False  False   
2      False     False   False    False  False     False  False  False   
3      False     False   False    False  False     False  False  False   
4      False     False   False    False  False     False  False  False   
...      ...       ...     ...      ...    ...       ...    ...    ...   
10836  False     False   False    False  False     False  False  False   
10837  False     False   False    False  False     False  False  False   
10838  False     False    True    False  False     False  False  False   
10839  False     False   False    False  False     False  False  False   
10840  False     False   False    False  False     False  False  False   

       content_rating  genres  last_updated  current_ver  android_ver  
0               False   False         False        False        False  
1               False   False         False        False        False  
2               False   False         False        False        False  
3               False   False         False        False        False  
4               False   False         False        False        False  
...               ...     ...           ...          ...          ...  
10836           False   False         False        False        False  
10837           False   False         False        False        False  
10838           False   False         False        False        False  
10839           False   False         False        False        False  
10840           False   False         False        False        False  

[10841 rows x 13 columns]

5. Проверьте нет ли пропусков после произведенной обработки

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   app             10841 non-null  object 
 1   category        10841 non-null  object 
 2   rating          9367 non-null   float64
 3   reviews         10841 non-null  object 
 4   size            10841 non-null  object 
 5   installs        10841 non-null  object 
 6   type            10840 non-null  object 
 7   price           10841 non-null  float64
 8   content_rating  10840 non-null  object 
 9   genres          10841 non-null  object 
 10  last_updated    10841 non-null  object 
 11  current_ver     10833 non-null  object 
 12  android_ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


6. Выведите на экран минимум и максимум из столбца `Price`:

7. Выведите на экран медиану и среднее арифметическое столбцов:
 - Rating
 - Reviews

8. Выведите на экран все уникальные значения категориального столбца Genres

9. Сгруппируйте данные по столбцу `Genres` и посчитайте для каждого жанра средний и медианный рейтинг. Представьте результат в виде нового DataFrame,который будет сохранен в переменную `grouped_df`, где одна колонка будет содержать названия жарнов, а две другие буду содержать в себе средний и медианный рейтинги. Напишите краткий вывод.

10. Из получившегося датафрейма выведите на экран только те жанры, чей медианный рейтинг больше 4.5. Напишите краткий вывод.

11. Выведете кол-во дубликатов в основном DataFrame (df) и удалите их 

In [ ]:
#Удаляем дубликаты
df.drop_duplicates(inplace=True)

12. Посчитатайте какое количество игр содержится в каждом жанре (value_counts). Выведите на экран топ-10 игр по количеству в жанре. Напишите краткий вывод.